balancer-js/src/modules/contracts/contracts.module.ts

```diff
@@ -1,78 +1,80 @@
+import {
+  AaveLinearPoolFactory,
+  AaveLinearPoolFactory__factory,
+  BalancerHelpers,
+  BalancerHelpers__factory,
+  ComposableStablePoolFactory,
+  ComposableStablePoolFactory__factory,
+  ERC20,
+  ERC20__factory,
+  ERC4626LinearPoolFactory,
+  ERC4626LinearPoolFactory__factory,
+  EulerLinearPoolFactory,
+  EulerLinearPoolFactory__factory,
+  GaugeClaimHelper,
+  GaugeClaimHelper__factory,
+  GearboxLinearPoolFactory,
+  GearboxLinearPoolFactory__factory,
+  LidoRelayer,
+  LidoRelayer__factory,
+  LiquidityGaugeV5,
+  LiquidityGaugeV5__factory,
+  RelayerV3,
+  RelayerV3__factory,
+  RelayerV4,
+  RelayerV4__factory,
+  RelayerV5,
+  RelayerV5__factory,
+  Vault,
+  Vault__factory,
+  WeightedPoolFactory,
+  WeightedPoolFactory__factory,
+  YearnLinearPoolFactory,
+  YearnLinearPoolFactory__factory,
+} from '@/contracts';
+import { Network } from '@/lib/constants/network';
+import { BALANCER_NETWORK_CONFIG } from '@/lib/constants/config';
+import { ContractAddresses } from '@/types';
 
 type ContractFactory = (
   address: string,
   signerOrProvider: Signer | Provider
 ) => Contract;
 
 export interface ContractInstances {
-  aaveLinearPoolFactory?: Contract;
+  aaveLinearPoolFactory?: AaveLinearPoolFactory;
   balancerHelpers: BalancerHelpers;
   BasePool: ContractFactory;
-  composableStablePoolFactory?: Contract;
+  composableStablePoolFactory?: ComposableStablePoolFactory;
   ERC20: ContractFactory;
-  erc4626LinearPoolFactory?: Contract;
-  eulerLinearPoolFactory?: Contract;
-  gaugeClaimHelper?: Contract;
-  gearboxLinearPoolFactory?: Contract;
+  erc4626LinearPoolFactory?: ERC4626LinearPoolFactory;
+  eulerLinearPoolFactory?: EulerLinearPoolFactory;
+  gaugeClaimHelper?: GaugeClaimHelper;
+  gearboxLinearPoolFactory?: GearboxLinearPoolFactory;
   lidoRelayer?: LidoRelayer;
   liquidityGauge: ContractFactory;
   multicall: Contract;
-  relayerV3?: Contract;
-  relayerV4?: Contract;
-  relayerV5?: Contract;
+  relayerV3?: RelayerV3;
+  relayerV4?: RelayerV4;
+  relayerV5?: RelayerV5;
   vault: Vault;
   veBal?: VeBal;
   veBalProxy?: VeBalProxy;
-  weightedPoolFactory?: Contract;
-  yearnLinearPoolFactory?: Contract;
+  weightedPoolFactory?: WeightedPoolFactory;
+  yearnLinearPoolFactory?: YearnLinearPoolFactory;
 }
 
 export class Contracts {
-  aaveLinearPoolFactory?: Contract;
-  balancerHelpers: BalancerHelpers;
-  composableStablePoolFactory?: Contract;
   contractAddresses: ContractAddresses;
-  erc4626LinearPoolFactory?: Contract;
-  eulerLinearPoolFactory?: Contract;
-  gaugeClaimHelper?: Contract;
-  gearboxLinearPoolFactory?: Contract;
-  lidoRelayer?: LidoRelayer;
-  multicall: Contract;
-  relayerV3?: Contract;
-  relayerV4?: Contract;
-  relayerV5?: Contract;
-  vault: Vault;
-  veBal?: VeBal;
-  veBalProxy?: VeBalProxy;
-  weightedPoolFactory?: Contract;
-  yearnLinearPoolFactory?: Contract;
+  private instances: ContractInstances;
```

Comment:
This looks way better to me 👏 
Let's see what everyone else thinks 🙃

Reply:
Thanks for the suggestion  😁